app/code/Magento/Deploy/Model/Filesystem.php

```diff
@@ -115,7 +115,7 @@ public function __construct(
         $this->userCollection = $userCollection;
         $this->locale = $locale;
         $this->functionCallPath =
-            PHP_BINARY . ' -f ' . BP . DIRECTORY_SEPARATOR . 'bin' . DIRECTORY_SEPARATOR . 'magento ';
+            '"' . PHP_BINARY . '"' . ' -f ' . BP . DIRECTORY_SEPARATOR . 'bin' . DIRECTORY_SEPARATOR . 'magento ';
```

Comment:
This is incomplete and will break if the path has double quotes in it. We'll also need to use `escapeshellcmd`.

Reply:
that's a much better alternative! `escapeshellcmd`